numpy/core/src/umath/scalarmath.c.src

```diff
@@ -1564,8 +1569,23 @@ static PyObject *
 
 
     val = PyArrayScalar_VAL(a, @Name@);
+    int retstatus = @name@_ctype_@oper@(val, &out);
 
-    @name@_ctype_@oper@(val, &out);
+    if (retstatus) {
+        int bufsize, errmask;
+        PyObject *errobj;
+
+        if (PyUFunc_GetPyValues("@name@_scalars", &bufsize, &errmask,
+                                &errobj) < 0) {
+            return NULL;
+        }
+        int first = 1;
+        if (PyUFunc_handlefperr(errmask, errobj, retstatus, &first)) {
+            Py_XDECREF(errobj);
+            return NULL;
+        }
+        Py_XDECREF(errobj);
+    }
```

Comment:
It is suspicious that these new lines are not marked as being tested

Reply:
Hmmm, right.  The first one is OK (I don't think it can be reasonably covered, it indicates an internal/critical error).

To hit the second one, we should also check for:
```
with np.errstate(over="raise"):
    with pytest.raises(FloatingPointError):
         operation
```